### PR TITLE
Feature/forms 468

### PR DESCRIPTION
**Pull request by TheSpartan1980**
# Description

Introduced another test form that utilises the majority of the components available. 

@jenbutongit:
- replaced node-sass with dart sass. linux distros don't seem to like node-sass. I tried rebuilding but still fails 
- move relevant dependencies in root to smoke-tests/designer/package.json 
- dockerfiles changed so they do not copy what they don't need 
- running yarn focus install to stop extraneous deps being installed

## Type of change

Please delete options that are not relevant.

- [x] Automated test

# How Has This Been Tested?

The test was created using the red-green-refactor cycle, the test has also been run in Chrome and Chrome headless locally

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts
- [ ] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

File: runner/src/server/forms/cg-test.json [hunk 1]
```diff
@@ -0,0 +1,397 @@
+{
```
Comment (gstevenson) [inline runner/src/server/forms/cg-test.json:1]: Is this file going to be permanent? We should probably call it something other than `cg-test` if so

File: smoke-tests/designer/features/actions/forms.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const { formRunner } = require("../pageobjects/pages");
+const formData = require("../../data/formData");
+
+class Forms {
+  cgTest() {
```
Comment (gstevenson) [inline smoke-tests/designer/features/actions/forms.js:5]: Similar comment to my first comment here

File: smoke-tests/designer/features/pageobjects/pages/formRunner.page.js [hunk 3]
```diff
@@ -2,10 +2,45 @@ class FormRunnerPage {
   open(path) {
     return browser.url(`${browser.config.runnerUrl}${path}/`);
   }
+
   get pageTitle() {
     return browser.$("h1");
   }
 
+  /**
```
Comment (gstevenson) [inline smoke-tests/designer/features/pageobjects/pages/formRunner.page.js:10]: nice docs 👍

File: smoke-tests/designer/features/actions/forms.js (hunk 2 above)
Comment (TheSpartan1980) [reply smoke-tests/designer/features/actions/forms.js]: @gstevenson Fixed

File: runner/src/server/forms/cg-test.json (hunk 1 above)
Comment (TheSpartan1980) [reply runner/src/server/forms/cg-test.json]: @gstevenson fixed

File: smoke-tests/designer/features/pageobjects/pages/formRunner.page.js [hunk 4]
```diff
@@ -2,10 +2,45 @@ class FormRunnerPage {
...
+      .$$(".govuk-input")
+      .findIndex((el) => el.parentElement().getText().includes(labelText));
+    return browser.$$(".govuk-input")[fieldIndex].setValue(fieldValue);
+  }
+
+  /**
+   * Inputs the address into the provided fields
+   * @param dataObject
+   */
+  ukAddress(dataObject) {
```
Comment (gstevenson) [inline smoke-tests/designer/features/pageobjects/pages/formRunner.page.js:37]: it looks like the other inputs on here (ie inputDate, inputField, selectRadio, selectFromList) have a verb based suffix, should we have one on here too so it's more obvious that it's going to enter the value?

File: smoke-tests/designer/features/pageobjects/pages/formRunner.page.js [hunk 5]
```diff
@@ -14,28 +49,54 @@ class FormRunnerPage {
...
+  get submitButton() {
     return browser.$(".govuk-button");
   }
 
   /**
    * Inputs text into a multiline text component and then clicks continue
    * @param text
    */
-  multilineText(text) {
+  textBox(text) {
```
Comment (gstevenson) [inline smoke-tests/designer/features/pageobjects/pages/formRunner.page.js:99]: And this one too actually, didn't spot it until I'd added the other comment

File: smoke-tests/designer/features/steps/completeAForm.steps.js [hunk 6]
```diff
@@ -35,5 +36,45 @@ Then("I taken directly to the page titled {string}", function (pageTitle) {
...
+    "Not supplied"
+  );
+  expect(formRunner.summaryAnswer(formData.multiLine.question)).toHaveText(
+    "I've turned it into a spaceship capable of interstellar travel"
+  );
+  expect(formRunner.summaryAnswer(formData.textField2.question)).toHaveText(
+    "Juan Pablo Montoya"
+  );
+  expect(formRunner.summaryAnswer(formData.emailAddress.question)).toHaveText(
+    "testing@testing.com"
```
Comment (gstevenson) [inline smoke-tests/designer/features/steps/completeAForm.steps.js:78]: On the absolute minor offchance we ever do anything with these emails like accidentally send out email notifications, it's safer to use `example.com`, `example.net` and `example.org` for test data - they're reserved domains that can be used for test data, whereas testing.com is a real address 😄

**Review by gstevenson — CHANGES_REQUESTED** [review]

File: smoke-tests/designer/features/steps/completeAForm.steps.js (hunk 6 above)
Comment (TheSpartan1980) [reply smoke-tests/designer/features/steps/completeAForm.steps.js]: @gstevenson fixed

File: smoke-tests/designer/features/pageobjects/pages/formRunner.page.js (hunk 5 above)
Comment (TheSpartan1980) [reply smoke-tests/designer/features/pageobjects/pages/formRunner.page.js]: @gstevenson fixed

File: smoke-tests/designer/features/pageobjects/pages/formRunner.page.js (hunk 4 above)
Comment (TheSpartan1980) [reply smoke-tests/designer/features/pageobjects/pages/formRunner.page.js]: @gstevenson fixed

**Review by gstevenson — APPROVED** [review]

**Pull request merged**